### DISCUSSION TITLE
Scatter text interpretation

### DISCUSSION THREAD

**Discussion by albin02t** [JasonKessler/scattertext#109] [Q&A]
Hi, First of the plot looks great and it is quite informative and intuitive.
But I'm having a hard time to interpret and understand some of the parameters as they aren't mentioned or touched on much in the README file and i get lost in the file.
So, if you don't mind can you explain:
1) How are the scores displayed on hover are calculated and what do they signify. I understand how the scaled f-scores are calculated each for the 2 classes.
2) What is the transform parameter used for and its values like st.Scalers.percentile_dense
3) term_scorer parameter and what does st.RankDifference() do 
4) sort_by_distance 
5) In the plot shown below, you can see that it is colored blue but score is a -0.015
![image](https://user-images.githubusercontent.com/69189799/167131731-626e2567-fa29-4dc4-9a97-e379cb80c46f.png)

**Comment by JasonKessler** [comment]
> How are the scores displayed on hover are calculated and what do they signify. I understand how the scaled f-scores are calculated each for the 2 classes.

The scores, unless the `term_scorer` parameter to `produce_scattertext_explorer` is used, correspond to the dense rank difference. 

> What is the transform parameter used for and its values like st.Scalers.percentile_dense
term_scorer parameter and what does st.RankDifference() do sort_by_distance

The `transform` parameter defines how a terms's frequency is transformed into an axis position. The simplest approach, `scale`, places the minimum frequency at the origin and the maximum frequency at (1,0) or (0,1).  The term_scorer determines the color of a point (dark blue at 1, and dark red at -1) as well as the score seen on hover.

`sort_by_distance` indicates whether the category-based term lists are based on the distance of each point to (0,1) or (1,0) (if True) or based on the scores computed by the `term_scorer`.

> In the plot shown below, you can see that it is colored blue but score is a -0.015

That's the dense rank difference.

**Reply by albin02t** [reply]
Really appreciate you reaching back so quick, but can you clarify a few more things.
1) How is the dense rank difference calculated?
2) Is RankDifference() different from dense rank difference and how?
3) If the term_scorer scores the terms as dark blue as 1 and dark red -1, then in the plot above the word 'goes'  is colored blue but the score is a -0.015. Could you explain how could that be?

Just starting out and thinking of implementing the scatter text in a product, so thanks for the help

**Comment by JasonKessler** [comment]
Please look at the source code and/or documentation for these questions.
